src/metal.zig

```diff
@@ -251,32 +262,118 @@ pub const Device = struct {
     }
 };
 
+pub const StreamingManager = struct {
+    const InflightBuffer = struct {
+        mtl_buffer: *mtl.Buffer,
+        queue: *Queue,
+        fence_value: u64,
+    };
+
+    mtl_device: *mtl.Device,
+    inflight_buffers: std.ArrayList(InflightBuffer),
+    free_buffers: std.ArrayList(*mtl.Buffer),
```

Comment:
nit: every `ArrayList` maintains a pointer to the allocator internally which can add up over time, we should probably replace all usages of `ArrayList` with `ArraylistUnmanaged` which take allocator as parameter to every method